libs/dal/src/repositories/base-repository.ts

```diff
@@ -88,12 +86,14 @@ export class BaseRepository<T_Query, T_Response> {
     return this.mapEntity(saved);
   }
 
-  async createMany(data: T_Query[]) {
-    await new Promise((resolve) => {
-      this.MongooseModel.collection.insertMany(data, (err, response) => {
-        resolve(response);
-      });
-    });
+  async createMany(data: T_Query[]): Promise<InsertManyResult<Document>> {
+    const saved = await this.MongooseModel.collection.insertMany(data);
+
+    return {
+      acknowledged: saved.acknowledged,
+      insertedCount: saved.insertedCount,
+      insertedIds: saved.insertedIds,
+    };
```

Comment:
The implementation wasn't working and during one of my trials to see how to upsert I implemented it. I am not using it at the end, I can leave it or remove it.